### PR TITLE
Re-enable FlatVector support for Decimal Type

### PR DESCRIPTION
**Pull request by kgpai**
Recently we had to revert #1572  (Add FlatVector Support for Decimal Type)  (revert commit : #1742) . This PR readds the commit . 
PR #1572   in combination with some other changes in an internal meta system triggered a regression affecting some systems. We have addressed the problem in our internal systems and are making changes to ensure we get better visibility before landing changes so that such incidents do not reccur. 

Here is the original commit message: 

```
Add support for Long and Short Decimal FlatVector.

1. Wrapped int128_t in a new value type LongDecimal.
2. Extended ByteStream::read function for LongDecimal for de-serialization.

Testing:
Added VectorTest::createDecimal tests that creates FlatVectors of Short and Long decimal.
Serialization and dictionary encoding are tested as part of this test.
```

### CODE REVIEW

File: velox/type/LongDecimal.h [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        static_cast<uint64_t>(value.unscaledValue() >> 64));
+    auto lowerHash =
+        folly::hasher<uint64_t>{}(static_cast<uint64_t>(value.unscaledValue()));
+    return facebook::velox::bits::hashMix(upperHash, lowerHash);
+  }
+};
+} // namespace folly
+
+namespace std {
+string to_string(facebook::velox::int128_t x);
```
Comment (majetideepak) [inline velox/type/LongDecimal.h:79]: Why is this required? Should this be `string to_string(LongDecimal x);`?

Comment (majetideepak) [reply velox/type/LongDecimal.h:79]: @Yuhta Jimmy added this in the original PR.

**Comment by majetideepak** [comment]
I think the final PR that landed was https://github.com/facebookincubator/velox/pull/1711
Nit: The title should probably be `Re-enable FlatVector support for Decimal Type`
@kgpai for this PR.

**Comment by kgpai** [comment]
Thanks Deepak, Updated title and description based on comment.

**Comment by majetideepak** [comment]
@kgpai you are right. Looks like @Yuhta squashed his changes in that PR. We can stick to https://github.com/facebookincubator/velox/pull/1572 in the description. Thanks for looking into this.

**Review by karteekmurthys — APPROVED** [review]

**Comment by kgpai** [comment]
Update: This should land before end of day.
